### PR TITLE
Added Qualcomm SnapDragon 820c support

### PR DESCRIPTION
**Pull request by IvanFarkas**
**/proc/device-tree/compatible:** arrow,apq8096-db820c^@qcom,apq8096-sbc^@
**/sys/firmware/devicetree/base/model:** Qualcomm Technologies, Inc. DB820c
**GPIO** ports from **[DragonBoard 820c Hardware Manual](https://github.com/sdrobertw/dragonboard820c/blob/master/hardware-docs/files/db820c-user-guide.docx)** (p16)

### CODE REVIEW

File: src/arm/96boards.c [hunk 1]
```diff
@@ -85,6 +96,99 @@ mraa_96boards_pininfo(mraa_board_t* board, int index, int sysfs_pin, char* fmt,
...
+			return MRAA_ERROR_NO_RESOURCES;
+		}
+	}
+	dev->mmap_write = &mraa_db820c_mmap_write;
+	dev->mmap_read = &mraa_db820c_mmap_read;
+	mmap_count++;
+
+	return MRAA_SUCCESS;
+}
+
```
Comment (Mani-Sadhasivam) [inline src/arm/96boards.c]: For the above code duplication, we can reuse the same mmap functions. But I haven't found the register documentation for this SoC yet. So, I would suggest you to remove mmap functions for now.

**Comment by Mani-Sadhasivam** [comment]
Also make the commit message to be more precise with your PR. It should portray what you are trying to achieve.

File: src/arm/96boards.c [hunk 2]
```diff
@@ -36,26 +36,37 @@
 
 #define DT_BASE "/proc/device-tree"
 
+#define PLATFORM_NAME_DB820C "DB820C"
 #define PLATFORM_NAME_DB410C "DB410C"
 #define PLATFORM_NAME_HIKEY "HIKEY"
 #define PLATFORM_NAME_BBGUM "BBGUM"
 #define MAX_SIZE 64
 #define MMAP_PATH "/dev/mem"
+#define DB820C_MMAP_REG 0x01000000
```
[the remark this answers is not in the document]
Comment (IvanFarkas) [reply src/arm/96boards.c]: Removed

File: src/arm/96boards.c [hunk 3]
```diff
@@ -36,26 +36,37 @@
...
 
+// Hikey
 int hikey_ls_gpio_pins[MRAA_96BOARDS_LS_GPIO_COUNT] = {
     488, 489, 490, 491, 492, 415, 463, 495, 426, 433, 427, 434,
 };
 
-
 const char* hikey_serialdev[MRAA_96BOARDS_LS_UART_COUNT] = { "/dev/ttyAMA2", "/dev/ttyAMA3" };
 
+// BbGum
```
[the remark this answers is not in the document]
Comment (IvanFarkas) [reply src/arm/96boards.c]: Updated

File: src/arm/96boards.c (hunk 1 above)
Comment (IvanFarkas) [reply src/arm/96boards.c]: Removed

File: src/arm/96boards.c [hunk 4]
```diff
@@ -139,55 +138,6 @@ mraa_db820c_mmap_read(mraa_gpio_context dev)
...
-			return MRAA_ERROR_NO_RESOURCES;
-		}
-	}
-	dev->mmap_write = &mraa_db820c_mmap_write;
-	dev->mmap_read = &mraa_db820c_mmap_read;
-	mmap_count++;
-
-	return MRAA_SUCCESS;
-}
-
```
Comment (Mani-Sadhasivam) [inline src/arm/96boards.c]: Still there are few 820c mmap functions. Please remove them also.
```
mraa_db820c_mmap_unsetup
mraa_db820c_mmap_read
mraa_db820c_mmap_write
```

**Comment by IvanFarkas** [comment]
I'd like MRAA to work on the DB820c SBC as well to implement IoT, IoMT functionality for sensors and actuators.

MRAA is dependent on [libsoc](https://github.com/jackmitch/libsoc).
I updated **[libsoc ](https://github.com/IvanFarkas/libsoc)** to work with DB820c SBC as well and submitted a [pull request](https://github.com/jackmitch/libsoc/pull/92) (pending).

Both libsoc and MRAA are carefully tested with the **[96Boards Sensor Mezzanine](https://github.com/96boards/Sensor_Mezzanine_Getting_Started)**.

**Comment by Mani-Sadhasivam** [comment]
> MRAA is dependent on libsoc.

This is not true. MRAA is not depending on any library.

File: src/arm/96boards.c [hunk 5]
```diff
@@ -85,6 +95,9 @@ mraa_96boards_pininfo(mraa_board_t* board, int index, int sysfs_pin, char* fmt,
     pininfo->gpio.mux_total = 0;
 }
 
+// TODO: DB820c
```
Comment (Mani-Sadhasivam) [inline src/arm/96boards.c]: Please remove this line. No need to mention this here

Comment (IvanFarkas) [reply src/arm/96boards.c]: Done

**Comment by IvanFarkas** [comment]
That's awesome. Thank you very much for your help and hard work.

File: src/arm/96boards.c [hunk 6]
```diff
@@ -36,26 +36,36 @@
...
+const char* db820c_serialdev[MRAA_96BOARDS_LS_UART_COUNT] = { "/dev/ttyMSM0", "/dev/ttyMSM1" };
+
+// DB410c
 int db410c_ls_gpio_pins[MRAA_96BOARDS_LS_GPIO_COUNT] = {
     36, 12, 13, 69, 115, 4, 24, 25, 35, 34, 28, 33,
 };
 
 const char* db410c_serialdev[MRAA_96BOARDS_LS_UART_COUNT] = { "/dev/ttyMSM0", "/dev/ttyMSM1" };
 
+// Hikey
```
Comment (alext-mkrs) [inline src/arm/96boards.c]: Please either don't do, or split out such styling fixes not related to the logical change in question (adding a board support), as they make it harder to analyze the git history.

There are several such ones, I won't repeat the comment fro all of them, for brevity, but please correct them all.

**Comment by IvanFarkas** [comment]
made changes

File: src/arm/96boards.c [hunk 7]
```diff
@@ -53,7 +60,6 @@ int hikey_ls_gpio_pins[MRAA_96BOARDS_LS_GPIO_COUNT] = {
     488, 489, 490, 491, 492, 415, 463, 495, 426, 433, 427, 434,
 };
 
-
```
Comment (alext-mkrs) [inline src/arm/96boards.c]: This is another surplus styling change I mentioned in the previous review. Please revert this.

File: src/arm/96boards.c [hunk 8]
```diff
@@ -199,7 +205,12 @@ mraa_96boards()
 
     if (mraa_file_exist(DT_BASE "/model")) {
         // We are on a modern kernel, great!!!!
-        if (mraa_file_contains(DT_BASE "/model", "Qualcomm Technologies, Inc. APQ 8016 SBC")) {
```
Comment (alext-mkrs) [inline src/arm/96boards.c]: As you've corrected the ordering in `arm.c` per @Mani-Sadhasivam's request, please do the same here, to use the same ordering style.

File: src/arm/96boards.c (hunk 7 above)
Comment (IvanFarkas) [reply src/arm/96boards.c]: done

**Comment by IvanFarkas** [comment]
All requested changes are done

**Comment by IvanFarkas** [comment]
Great. Thx, Ivan

**Comment by malikabhi05** [comment]
@IvanFarkas I am planning to merge you commits, but it looks like one of the commits is not signed off, would you like to quickly do that?

**Comment by IvanFarkas** [comment]
It is.
```
Signed-off-by: Ivan Farkas <Ivan.Farkas@TeleCuris.com>
```

**Comment by IvanFarkas** [comment]
Done

**Comment by malikabhi05** [comment]
Merged, Thanks Ivan!

**Pull request closed without merging**